node/src/blockchain/raw_transaction.rs

```diff
@@ -90,55 +92,192 @@ mod tests {
+        let first_part_tx_2 = r#"[{"nonce": "0x0", "gasPrice": "0xd55698372431", "gasLimit": "0x1e8480", "to": "0xF0109fC8DF283027b6285cc889F5aA624EaC1F55", "value": "0x3b9aca00", "data": []}, {"private_key": "0x4c0883a69102937d6231471b5dbb6204fe5129617082792ae468d01a3f362318", "signed": "#;
+        let first_part_tx_3 = r#"[{"nonce": "0x00", "gasPrice": "0x09184e72a000", "gasLimit": "0x2710", "to": null, "value": "0x00", "data": [127,116,101,115,116,50,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,96,0,87]}, {"private_key": "0xe331b6d69882b4cb4ea581d88e0b604039a3de5967688d3dcffdd2270c0fd109", "signed": "#;
+        fn compose(first_part: &str, slice: &[u8]) -> String {
+            let third_part_jrc = "}]";
+            format!("{}{:?}{}", first_part, slice, third_part_jrc)
+        }
+        let all_transactions = format!(
+            "[{}]",
+            vec![first_part_tx_1, first_part_tx_2, first_part_tx_3]
+                .iter()
+                .zip(collection_of_tx_signatures.iter())
+                .zip(0usize..2)
+                .fold(String::new(), |so_far, actual| [
+                    so_far,
+                    compose(actual.0 .0, actual.0 .1)
+                ]
+                .join(if actual.1 == 0 { "" } else { ", " }))
+        );
+        let txs: Vec<(RawTransaction, Signing)> = serde_json::from_str(&all_transactions).unwrap();
+        let constant_parts = &[
+            &[
+                248u8, 108, 9, 133, 4, 168, 23, 200, 0, 130, 82, 8, 148, 53, 53, 53, 53, 53, 53,
+                53, 53, 53, 53, 53, 53, 53, 53, 53, 53, 53, 53, 53, 53, 136, 13, 224, 182, 179,
+                167, 100, 0, 0, 128,
+            ][..],
+            &[
+                248, 106, 128, 134, 213, 86, 152, 55, 36, 49, 131, 30, 132, 128, 148, 240, 16, 159,
+                200, 223, 40, 48, 39, 182, 40, 92, 200, 137, 245, 170, 98, 78, 172, 31, 85, 132,
+                59, 154, 202, 0, 128,
+            ][..],
+            &[
+                248, 117, 128, 134, 9, 24, 78, 114, 160, 0, 130, 39, 16, 128, 128, 164, 127, 116,
+                101, 115, 116, 50, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                0, 0, 0, 0, 0, 0, 96, 0, 87,
+            ][..],
+        ];
+        let lengths_of_constant_parts: Vec<usize> =
+            constant_parts.iter().map(|part| part.len()).collect();
+        for (((tx, signed), length), constant_part) in txs
+            .iter()
+            .zip(lengths_of_constant_parts)
+            .zip(constant_parts)
+        {
+            let sign = tx.sign(
+                &Wallet::from(
+                    Bip32ECKeyPair::from_raw_secret(&signed.private_key.0.as_ref()).unwrap(),
+                ),
+                chain,
             );
+            assert_eq!(sign, signed.signed);
+            assert_eq!(sign[..length], **constant_part)
         }
     }
 
+    //TODO I think the tests below just duplicate what the other tests already tested
```

Comment:
The distinction with these additional tests help to ensure the code creates valid signed transactions even with various chain IDs especially since Polygon Mumbai's ID is such a large number.

Reply:
I can understand the general purpose of the tests in this file but I struggle to understand why it speaks of seemingly two situations with different kind of transactions (well it was more obvious before I changed the first sections of the tests - so please look at the master branch to see it unbiased) there have been tests like:

"test_signs_transaction_ropsten()" 
and later on another one called as 
"test_transfer_transaction_signing_ropsten"

In my eyes they test the pretty much same thing. But I'm asking for confirming it. 
I'd like to get rid of the last two (maybe extra) tests if you agree.